politeiawww/legacy/comments/process.go

```diff
@@ -314,20 +314,36 @@ func (c *Comments) processVotes(ctx context.Context, v v1.Votes) (*v1.VotesReply
 	cv := convertCommentVotes(votes)
 
 	// Populate comment votes with user data
-	uid, err := uuid.Parse(v.UserID)
-	if err != nil {
-		return nil, err
+	var uids []uuid.UUID
+	if v.UserID != "" {
+		// If user ID filter is applied, we have only one user
+		// to fetch.
+		uid, err := uuid.Parse(v.UserID)
+		if err != nil {
+			return nil, err
+		}
+		uids = append(uids, uid)
+	} else {
+		// If user ID filter is not applied, we need to collect all
+		// the user IDs from comment vote structs.
+		for _, vote := range votes {
+			uid, err := uuid.Parse(vote.UserID)
+			if err != nil {
+				return nil, err
+			}
+			uids = append(uids, uid)
```

Comment:
should avoid adding duplicate uids

Reply:
https://github.com/decred/politeia/pull/1598/commits/c8e7342ec7a65d50280e1ab5f1d2045a03e0cbec